.github/workflows/ci.yml

```diff
@@ -58,6 +58,28 @@ jobs:
       - name: Run Clippy
         run: cargo clippy --all-targets --target ${{ matrix.target }}
 
+  miri:
+    name: "miri"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Miri
+        run: |
+          MIRI_NIGHTLY=nightly-$(curl -s https://rust-lang.github.io/rustup-components-history/x86_64-unknown-linux-gnu/miri)
+          echo "Installing latest nightly with Miri: $MIRI_NIGHTLY"
+          rustup default "$MIRI_NIGHTLY"
+          rustup component add miri
```

Comment:
Would something like `rustup toolchain install nightly --component miri; rustup default nightly` work? I believe `rustup toolchain install` will fallback to older nightly versions if a component is unavailable.

Reply:
Maybe? The custom Miri logic predates that rustup feature and it is hard to test.

I know that the script I used here definitely works. :)
Though I guess I could refrain from fixing the Miri toolstate today, and then we can test this tomorrow.

Reply:
That is so much more complicated and verbose than the alternatives that I don't quite see why one would use this approach?

Reply:
Yes that new snippet seems to work fine. :-)